packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
I disagree. The spec broke a common rule, and as long as we allow declarations without an id we [are](https://github.com/babel/babel/blob/d82afb407eaa4ff03b98b85824efcd161913952d/packages/babel-traverse/src/path/introspection.js#L290) [going](https://github.com/babel/babel/blob/d82afb407eaa4ff03b98b85824efcd161913952d/packages/babel-plugin-transform-es2015-block-scoped-functions/src/index.js#L7-11) [to](https://github.com/babel/babili/blob/6f0c1dcf6ce2af602d718d4ad93135a34ad7ca59/packages/babel-plugin-minify-dead-code-elimination/src/index.js#L262-L276) [have](https://github.com/babel/babili/blob/6f0c1dcf6ce2af602d718d4ad93135a34ad7ca59/packages/babel-plugin-minify-dead-code-elimination/src/index.js#L983-L990) [bugs](https://github.com/babel/babel/blob/d82afb407eaa4ff03b98b85824efcd161913952d/packages/babel-helper-remap-async-to-generator/src/index.js#L146-L149).